e2e/test/DeviceClientX509AuthenticationE2ETests.cs

```diff
@@ -0,0 +1,160 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public class DeviceClientX509AuthenticationE2ETests : IDisposable
+    {
+        private static readonly string DevicePrefix = $"E2E_{nameof(DeviceClientX509AuthenticationE2ETests)}_";
+        private static readonly TestLogging _log = TestLogging.GetInstance();
+        private static readonly TimeSpan TIMESPAN_ONE_MINUTE = TimeSpan.FromMinutes(1);
+        private static readonly TimeSpan TIMESPAN_ONE_SECOND = TimeSpan.FromSeconds(1);
+        private static readonly TimeSpan TIMESPAN_FIVE_SECONDS = TimeSpan.FromSeconds(5);
+        private static readonly TimeSpan TIMESPAN_TWENDY_SECONDS = TimeSpan.FromSeconds(20);
+
+        private readonly ConsoleEventListener _listener;
+        private readonly string _hostName;
+
+        public DeviceClientX509AuthenticationE2ETests()
+        {
+            _listener = TestConfig.StartEventListener();
+            _hostName = TestDevice.GetHostName(Configuration.IoTHub.ConnectionString);
+        }
+
+        
+        [TestMethod]
+        public async Task X509_InvalidDeviceId_Amqp()
+        {
+            await X509InvalidDeviceIdOpenAsyncTest(Client.TransportType.Amqp).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task X509_InvalidDeviceId_Amqp_Tcp()
+        {
+            await X509InvalidDeviceIdOpenAsyncTest(Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task X509_InvalidDeviceId_Amqp_WebSocket()
```

Comment:
Should the test names be more specific to the operation it's performing underneath? That will be more readable for all of these tests.